scripts/zones/Lower_Jeuno/npcs/Treasure_Coffer.lua

```diff
@@ -478,6 +590,77 @@ local function givePrize(player, ki)
     end
 end
 
+
+function addonGear(player, option, addonReward)
+    local param1     = 0 -- Placement of augment 1 table
+    local param2     = 0 -- Placement of augment 2 table
+    local param3     = 0 -- Leftover bits
+    local gear       = 0 -- Gear selected
+    local addon      = 0 -- ACP, AMK, ASA
+    local parameter1 = optionToAugment[addon][param1] -- Gathering table augments
+    local parameter2 = optionToAugment[addon][param2] -- Gathering table augments
+    local addAug     = {}
+
+    param3   = bit.rshift(option, 21)                                                      -- These bits are lefover. Non critial to implement
+    param2   = bit.rshift(option, 16) - (param3 * 32)                                      -- Second selected augment option
+    param1   = bit.rshift(option, 11) - (param3 * 1024) - (param2 * 32)                    -- First selected augment option
+    gear     = bit.rshift(option, 6)  - (param3 * 32768)- (param2 * 1024) - (param1 * 32)  -- Gear selection
+    addon    = optionToGear[gear].addon
+
+    if addonReward then
+        option = option  - 8388615 -- extra bit passed when recieving armor
+    end
+
+    if addonReward then
+        for i = 1, #parameter1 do                        -- Adding parameter1 augments table
+            table.insert(addAug, parameter1[i].augment)
+            table.insert(addAug, parameter1[i].power)
+        end
+        for i = 1, #parameter2 do                        -- Adding parameter2 augments table
+            table.insert(addAug, parameter2[i].augment)
+            table.insert(addAug, parameter2[i].power)
+        end
+
+        if player:getFreeSlotsCount() == 0 then
+            player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED, optionToGear[gear].itemid)
+        else
+            player:addItem(optionToGear[gear].itemid, 1, unpack(addAug))
+            player:messageSpecial(ID.text.ITEM_OBTAINED, optionToGear[gear].itemid)
+            player:delKeyItem(({xi.ki.PRISMATIC_KEY, xi.ki.OXBLOOD_KEY, xi.ki.BEHEMOTH_KEY})[addon])
+        end
+    else
+        for i = 1, #parameter1 do -- Converting augment selection 1 integer values into binary for updateEvent (5 bits for power followed by 11 bits for ID)
+            table.insert(addAug, string.format("%05i%011i", intToBinary(parameter1[i].power), intToBinary(parameter1[i].augment)))
+        end
+        for i = 1, #parameter2 do -- Converting augment selection 2 integer values into binary for updateEvent (5 bits for power followed by 11 bits for ID)
+            table.insert(addAug, string.format("%05i%011i", intToBinary(parameter2[i].power), intToBinary(parameter2[i].augment)))
+        end
+        for i = #addAug, 5 do
+            table.insert(addAug, "0000000000000000")
```

Comment:
ProTip: Teo despises `table.insert`s especially for usage. We should not be on the fly modifying tables to accomplish any of this.